### ISSUE TITLE
[plugin css-file] entry file not found

### ISSUE THREAD

**Issue by IgnisDa** [remix-run/remix#4726]
### What version of Remix are you using?

1.8.0

### Steps to Reproduce

1. Clone this repo: https://github.com/IgnisDa/npm-libs
2. `pnpm install`
3. `pnpm nx run remix-pagination-demo:build`

Note: the styles work fine in dev mode.

### Expected Behavior

Build to be completed successfully.

### Actual Behavior

Here is the logs of a recent remix build on GH actions:

https://github.com/IgnisDa/npm-libs/actions/runs/3588161240/jobs/6039249113

Logs:

```
> cross-env NODE_ENV=production remix build

Building Remix app in production mode...
The following error is a bug in Remix; please open an issue! https://github.com/remix-run/remix/issues/new
The following error is a bug in Remix; please open an issue! https://github.com/remix-run/remix/issues/new
The following error is a bug in Remix; please open an issue! https://github.com/remix-run/remix/issues/new
The following error is a bug in Remix; please open an issue! https://github.com/remix-run/remix/issues/new
The following error is a bug in Remix; please open an issue! https://github.com/remix-run/remix/issues/new
The following error is a bug in Remix; please open an issue! https://github.com/remix-run/remix/issues/new
The following error is a bug in Remix; please open an issue! https://github.com/remix-run/remix/issues/new
The following error is a bug in Remix; please open an issue! https://github.com/remix-run/remix/issues/new

✘ [ERROR] [plugin css-file] entry file not found

    ../../node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/compiler/plugins/cssFilePlugin.js:125:28:
      125 │         invariant["default"](entryFile, "entry file not found");
          ╵                             ^

    at Object.invariant [as default] (/workspace/node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/invariant.js:18:11)
    at /workspace/node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/compiler/plugins/cssFilePlugin.js:125:29
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async requestCallbacks.on-load (/workspace/node_modules/.pnpm/esbuild@0.15.12/node_modules/esbuild/lib/main.js:1388:22)
    at async handleRequest (/workspace/node_modules/.pnpm/esbuild@0.15.12/node_modules/esbuild/lib/main.js:697:13)

  This error came from the "onLoad" callback registered here:

    ../../node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/compiler/plugins/cssFilePlugin.js:58:12:
      58 │       build.onLoad({
         ╵             ~~~~~~

    at setup (/workspace/node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/compiler/plugins/cssFilePlugin.js:58:13)
    at handlePlugins (/workspace/node_modules/.pnpm/esbuild@0.15.12/node_modules/esbuild/lib/main.js:1251:21)
    at buildOrServeImpl (/workspace/node_modules/.pnpm/esbuild@0.15.12/node_modules/esbuild/lib/main.js:942:5)
    at Object.buildOrServe (/workspace/node_modules/.pnpm/esbuild@0.15.12/node_modules/esbuild/lib/main.js:750:5)
    at /workspace/node_modules/.pnpm/esbuild@0.15.12/node_modules/esbuild/lib/main.js:2085:17
    at new Promise (<anonymous>)
    at Object.build (/workspace/node_modules/.pnpm/esbuild@0.15.12/node_modules/esbuild/lib/main.js:2084:14)
    at Object.build (/workspace/node_modules/.pnpm/esbuild@0.15.12/node_modules/esbuild/lib/main.js:1931:51)
    at Object.compile (/workspace/node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/compiler/compileBrowser.js:117:43)
    at Object.compile (/workspace/node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/compiler/remixCompiler.js:28:43)
    at Object.build (/workspace/node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/compiler/build.js:33:23)
    at Object.build (/workspace/node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/cli/commands.js:169:17)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Object.run (/workspace/node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/cli/run.js:448:7)

  The plugin "css-file" was triggered by this import

    app/root.tsx:10:19:
      10 │ import styles from './tailwind.css';
         ╵                    ~~~~~~~~~~~~~~~~

 ✘ [ERROR] [plugin css-file] entry file not found

    ../../node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/compiler/plugins/cssFilePlugin.js:125:28:
      125 │         invariant["default"](entryFile, "entry file not found");
          ╵                             ^

    at Object.invariant [as default] (/workspace/node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/invariant.js:18:11)
    at /workspace/node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/compiler/plugins/cssFilePlugin.js:125:29
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async requestCallbacks.on-load (/workspace/node_modules/.pnpm/esbuild@0.15.12/node_modules/esbuild/lib/main.js:1388:22)
    at async handleRequest (/workspace/node_modules/.pnpm/esbuild@0.15.12/node_modules/esbuild/lib/main.js:697:13)

  This error came from the "onLoad" callback registered here:

    ../../node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/compiler/plugins/cssFilePlugin.js:58:12:
      58 │       build.onLoad({
         ╵             ~~~~~~

    at setup (/workspace/node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/compiler/plugins/cssFilePlugin.js:58:13)
    at handlePlugins (/workspace/node_modules/.pnpm/esbuild@0.15.12/node_modules/esbuild/lib/main.js:1251:21)
    at buildOrServeImpl (/workspace/node_modules/.pnpm/esbuild@0.15.12/node_modules/esbuild/lib/main.js:942:5)
    at Object.buildOrServe (/workspace/node_modules/.pnpm/esbuild@0.15.12/node_modules/esbuild/lib/main.js:750:5)
    at /workspace/node_modules/.pnpm/esbuild@0.15.12/node_modules/esbuild/lib/main.js:2085:17
    at new Promise (<anonymous>)
    at Object.build (/workspace/node_modules/.pnpm/esbuild@0.15.12/node_modules/esbuild/lib/main.js:2084:14)
    at Object.build (/workspace/node_modules/.pnpm/esbuild@0.15.12/node_modules/esbuild/lib/main.js:1931:51)
    at Object.compile (/workspace/node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/compiler/compileBrowser.js:117:43)
    at Object.compile (/workspace/node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/compiler/remixCompiler.js:28:43)
    at Object.build (/workspace/node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/compiler/build.js:33:23)
    at Object.build (/workspace/node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/cli/commands.js:169:17)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Object.run (/workspace/node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/cli/run.js:448:7)

  The plugin "css-file" was triggered by this import

    app/routes/index.tsx:3:28:
      3 │ import indexPageStyles from '../styles/pages/index.css';
        ╵                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~

 ✘ [ERROR] [plugin css-file] entry file not found

    ../../node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/compiler/plugins/cssFilePlugin.js:125:28:
      125 │         invariant["default"](entryFile, "entry file not found");
          ╵                             ^

    at Object.invariant [as default] (/workspace/node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/invariant.js:18:11)
    at /workspace/node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/compiler/plugins/cssFilePlugin.js:125:29
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async requestCallbacks.on-load (/workspace/node_modules/.pnpm/esbuild@0.15.12/node_modules/esbuild/lib/main.js:1388:22)
    at async handleRequest (/workspace/node_modules/.pnpm/esbuild@0.15.12/node_modules/esbuild/lib/main.js:697:13)

  This error came from the "onLoad" callback registered here:

    ../../node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/compiler/plugins/cssFilePlugin.js:58:12:
      58 │       build.onLoad({
         ╵             ~~~~~~

    at setup (/workspace/node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/compiler/plugins/cssFilePlugin.js:58:13)
    at handlePlugins (/workspace/node_modules/.pnpm/esbuild@0.15.12/node_modules/esbuild/lib/main.js:1251:21)
    at buildOrServeImpl (/workspace/node_modules/.pnpm/esbuild@0.15.12/node_modules/esbuild/lib/main.js:942:5)
    at Object.buildOrServe (/workspace/node_modules/.pnpm/esbuild@0.15.12/node_modules/esbuild/lib/main.js:750:5)
    at /workspace/node_modules/.pnpm/esbuild@0.15.12/node_modules/esbuild/lib/main.js:2085:17
    at new Promise (<anonymous>)
    at Object.build (/workspace/node_modules/.pnpm/esbuild@0.15.12/node_modules/esbuild/lib/main.js:2084:14)
    at Object.build (/workspace/node_modules/.pnpm/esbuild@0.15.12/node_modules/esbuild/lib/main.js:1931:51)
    at Object.compile (/workspace/node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/compiler/compileBrowser.js:117:43)
    at Object.compile (/workspace/node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/compiler/remixCompiler.js:28:43)
    at Object.build (/workspace/node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/compiler/build.js:33:23)
    at Object.build (/workspace/node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/cli/commands.js:169:17)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Object.run (/workspace/node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/cli/run.js:448:7)

  The plugin "css-file" was triggered by this import

    app/routes/pokemon.tsx:7:30:
      7 │ import pokemonPageStyles from '../styles/pages/pokemon.css';
        ╵                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

 ✘ [ERROR] [plugin css-file] entry file not found

    ../../node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/compiler/plugins/cssFilePlugin.js:125:28:
      125 │         invariant["default"](entryFile, "entry file not found");
          ╵                             ^

    at Object.invariant [as default] (/workspace/node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/invariant.js:18:11)
    at /workspace/node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/compiler/plugins/cssFilePlugin.js:125:29
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async requestCallbacks.on-load (/workspace/node_modules/.pnpm/esbuild@0.15.12/node_modules/esbuild/lib/main.js:1388:22)
    at async handleRequest (/workspace/node_modules/.pnpm/esbuild@0.15.12/node_modules/esbuild/lib/main.js:697:13)

  This error came from the "onLoad" callback registered here:

    ../../node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/compiler/plugins/cssFilePlugin.js:58:12:
      58 │       build.onLoad({
         ╵             ~~~~~~

    at setup (/workspace/node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/compiler/plugins/cssFilePlugin.js:58:13)
    at handlePlugins (/workspace/node_modules/.pnpm/esbuild@0.15.12/node_modules/esbuild/lib/main.js:1251:21)
    at buildOrServeImpl (/workspace/node_modules/.pnpm/esbuild@0.15.12/node_modules/esbuild/lib/main.js:942:5)
    at Object.buildOrServe (/workspace/node_modules/.pnpm/esbuild@0.15.12/node_modules/esbuild/lib/main.js:750:5)
    at /workspace/node_modules/.pnpm/esbuild@0.15.12/node_modules/esbuild/lib/main.js:2085:17
    at new Promise (<anonymous>)
    at Object.build (/workspace/node_modules/.pnpm/esbuild@0.15.12/node_modules/esbuild/lib/main.js:2084:14)
    at Object.build (/workspace/node_modules/.pnpm/esbuild@0.15.12/node_modules/esbuild/lib/main.js:1931:51)
    at Object.compile (/workspace/node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/compiler/compileBrowser.js:117:43)
    at Object.compile (/workspace/node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/compiler/remixCompiler.js:28:43)
    at Object.build (/workspace/node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/compiler/build.js:33:23)
    at Object.build (/workspace/node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/cli/commands.js:169:17)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Object.run (/workspace/node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/cli/run.js:448:7)

  The plugin "css-file" was triggered by this import

    ../../packages/remix-pagination/src/index.ts:2:34:
      2 │ export { default as styles } from './lib/index.module.css';
        ╵                                   ~~~~~~~~~~~~~~~~~~~~~~~~


Build failed with 4 errors:
../../node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/compiler/plugins/cssFilePlugin.js:125:28: ERROR: [plugin: css-file] entry file not found
../../node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/compiler/plugins/cssFilePlugin.js:125:28: ERROR: [plugin: css-file] entry file not found
../../node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/compiler/plugins/cssFilePlugin.js:125:28: ERROR: [plugin: css-file] entry file not found
../../node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/compiler/plugins/cssFilePlugin.js:125:28: ERROR: [plugin: css-file] entry file not found
Error
    at Object.onCompileFailure (/workspace/node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/cli/commands.js:174:13)
    at Object.compile (/workspace/node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/compiler/remixCompiler.js:33:134)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Object.build (/workspace/node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/compiler/build.js:33:3)
    at async Object.build (/workspace/node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/cli/commands.js:169:3)
    at async Object.run (/workspace/node_modules/.pnpm/@remix-run+dev@1.8.0_swfla57xeclfb55c5rcsqyoffe/node_modules/@remix-run/dev/dist/cli/run.js:448:7)
 ELIFECYCLE  Command failed with exit code 1.
 WARN   Local package.json exists, but node_modules missing, did you mean to install?
ERROR: "build:remix" exited with 1.
 ELIFECYCLE  Command failed with exit code 1.
 WARN   Local package.json exists, but node_modules missing, did you mean to install?

 >  NX   ERROR: Something went wrong in run-commands - Command failed: pnpm run build

   Pass --verbose to see the stacktrace.


 ———————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

 >  NX   Ran target build for project remix-pagination-demo and 2 task(s) it depends on (9s)
 
    ✖    1/3 failed
    ✔    2/3 succeeded [1 read from cache]
```

**Comment by Coobaha** [comment]
I had the same issue; it was due to cssFilePlugin not finding an entry path that is relative and contains `dots`

```typescript
// normalizedEntry, in my case, was '../server`

let entryFile = outputFiles.find(file => {
   return normalizePathSlashes(file.path).endsWith(normalizedEntry);
});
```

Our server lives in the parent folder, and `remix build/watch/dev` outputs to the parent folder.


As a workaround `remix app` has a `server` as a symlink

**Comment by marwan38** [comment]
Facing the same issue. Don't have a solution just yet.

**Comment by egorpavlikhin** [comment]
I was trying the latest remix version (`1.13.0`) with experimental tailwind support and hit the same issue. The code changes in `cssFilePlugin.js` from @marwan38 fixed the issue for me.

**Comment by markdalgleish** [comment]
@IgnisDa I think this issue was fixed by https://github.com/remix-run/remix/pull/4855, released in v1.10.0. Once updated, can you confirm whether you're still experiencing the issue? Note that in your linked repo you might need to rename your `.module.css` files to just `.css` since newer versions of Remix will treat these as CSS Modules. I'm closing this issue since I'm pretty sure this is resolved, but feel free to re-open if you're still having problems from v1.10.0 upwards.

@egorpavlikhin I'm surprised you hit this issue with v1.13.0. Are you sure it's the same issue? If you're still having trouble, could you share a minimal repro? If it's a different issue, could you open a separate issue on GitHub?

**Comment by egorpavlikhin** [comment]
@markdalgleish I am still hitting this issue in v2 if I drop the patchfile, let me try to make a repro.
